src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -256,6 +260,36 @@ public void start(Map<String, String> properties) {
     );
   }
 
+  protected Map<String, Object> computeInitialOffset(
+          String tableOrQuery,
+          Map<String, Object> partitionOffset,
+          TimeZone timezone) {
+    if (!(partitionOffset == null)) {
+      return partitionOffset;
+    } else {
+      Map<String, Object> initialPartitionOffset = null;
+      // no offsets found
+      Long timestampInitial = config.getLong(JdbcSourceConnectorConfig.TIMESTAMP_INITIAL_CONFIG);
+      if (timestampInitial != null) {
+        // start at the specified timestamp
+        if (timestampInitial == JdbcSourceConnectorConfig.TIMESTAMP_INITIAL_CURRENT) {
+          // use the current time
+          try (Connection con = cachedConnectionProvider.getConnection()) {
```

Comment:
Can you pull the `getConnection` outside of the try-with-resources?
It's a cached connection object, and closing it here invalidates that object elsewhere and forces the task to reconnect before continuing. See how `validateNonNullable` does it for an example.

Reply:
got it, I moved this one. Could you check my change if I understand your comment ?